src/Symfony/Component/Notifier/Bridge/Isendpro/IsendproTransport.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Isendpro;
+
+use Symfony\Component\HttpClient\Exception\JsonException;
+use Symfony\Component\HttpClient\Exception\TransportException as HttpClientTransportException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Component\Uid\Uuid;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class IsendproTransport extends AbstractTransport
+{
+    protected const HOST = 'apirest.isendpro.com';
+
+    private ?string $from = null;
+    private bool $noStop = false;
+    private bool $sandbox = false;
+
+    public function __construct(
+        #[\SensitiveParameter] private string $keyid,
+        HttpClientInterface $client = null,
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (null === $this->from) {
+            return sprintf('isendpro://%s?no_stop=%s&sandbox=%s', $this->getEndpoint(), (int) $this->noStop, (int) $this->sandbox);
+        }
+
+        return sprintf('isendpro://%s?from=%s&no_stop=%s&sandbox=%s', $this->getEndpoint(), $this->from, (int) $this->noStop, (int) $this->sandbox);
+    }
+
+    public function setFrom(?string $from): static
+    {
+        $this->from = $from;
+
+        return $this;
+    }
+
+    public function setNoStop(bool $noStop): static
+    {
+        $this->noStop = $noStop;
+
+        return $this;
+    }
+
+    public function setSandbox(bool $sandbox): static
+    {
+        $this->sandbox = $sandbox;
+
+        return $this;
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $messageId = Uuid::v4()->toRfc4122();
```

Comment:
Maybe microtime or anything which can be used without symfony/uid is enough.

I would avoid the dependency here